expirationwatch/expiration_watcher_test.go

```diff
@@ -123,38 +125,44 @@ func TestRemoveItemWhichSharesExpirationTimeWithOtherItems(t *testing.T) {
 	assert.Equal(t, expiryEntryOne, pruned[0])
 }
 
-func TestStartsAndStopsPoller(t *testing.T) {
+func TestStartAndStopPoller(t *testing.T) {
 	watcher := New(0)
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
Since we are explicitly calling cancel below, do we need to defer calling it as well?

Reply:
Yes, because the function could return (e.g. if any `assert.Require` fails) before we reach the explicit call to `cancel`, which would result in a leaked goroutine.